Add recipe for ue-mode

manenko opened the pull request:
### Brief summary of what the package does

The `ue.el` provides a minor mode for working with Unreal Engine projects in GNU Emacs.

### Direct link to the package repository

https://gitlab.com/unrealemacs/ue.el

### Your association with the package

I am a maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented on recipes/ue:
```diff
@@ -0,0 +1,3 @@
+(ue :repo    "unrealemacs/ue.el"
+    :fetcher gitlab
+    :files   ("*.el" "snippets"))
```
I believe this is equivalent (and gives your recipe room to grow without another PR):
```suggestion
    :files   (:defaults "snippets"))
```

riscy reviewed (commented):
Thanks for this!  Just a quick first pass over the checklist --

### ue.el ###

byte-compile (using Emacs 27.1):
```
ue.el:380:19:Warning: reference to free variable `yas-snippet-dirs'
ue.el:380:19:Warning: assignment to free variable `yas-snippet-dirs'
In ue--register-keywords:
ue.el:404:9:Warning: assignment to free variable `ue--keywords-registered'
In end of data:
ue.el:725:1:Warning: the following functions are not known to be defined:
    yas-load-directory, yas-activate-extra-mode
```
Possibly you need a Package-Requires/`require` on yasnippet?  If this is meant to be optional, you could use `defvar` and `declare-function` to pacify the byte compiler.

package-lint (using version 20210528.2348):
```
1 issue found:
719:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
```
It's probably best to do this step when a user first enables the mode, to minimize side effects from merely loading in your package.

Other [lints](https://github.com/riscy/melpazoid):
- ue.el#L55: It's safer to sharp-quote function names; use `#'`

manenko commented:
Done! 

Thank you for your review and comments! I've seen a warning about `with-eval-after-load`, but it seems every package that provides custom snippets uses this approach to activate them, so I thought it was legit. Anyway, I changed it according to your suggestion as it is more logical.

As for byte-compile issues - my Emacs 27.2 shows no warnings :O. I fixed them too but cannot validate this.

riscy commented:
> I've seen a warning about with-eval-after-load, but it seems every package that provides custom snippets uses this approach to activate them, so I thought it was legit.

To be fair, I've been wrong about this in the past and some exceptions to the "rule" include adding to `auto-mode-alist`, or `auth-source-backend-parser-functions`.  On the bright side, if I'm wrong, it's not a breaking change for your package.  Whereas packages that were eager about making these modifications might break user configurations if they go back.

In any case, welcome to MELPA. :)

manenko commented:
Thank you! :)
